src/components/shared/Icon.js

```diff
@@ -6,14 +6,17 @@
 
 import React, { PureComponent } from 'react';
 import explicitConnect from '../../utils/connect';
-import { getIconClassNameForCallNode } from '../../selectors/icons';
+import { getIconClassName } from '../../selectors/icons';
 import { iconStartLoading } from '../../actions/icons';
 
 import type { CallNodeDisplayData } from '../../types/profile-derived';
 import type { ConnectedProps } from '../../utils/connect';
 
 type OwnProps = {|
-  +displayData: CallNodeDisplayData,
+  // This prop is used by call tree.
+  +displayData?: CallNodeDisplayData,
+  // This prop is for other parts of profiler.
+  +iconUrl?: string | null,
```

Comment:
Reusing the NodeIcon component but changed some of the logic to make it more generic. Also changed its name, since it's no longer a NodeIcon only :)